gradio/components/annotated_image.py

```diff
@@ -141,12 +144,12 @@ def postprocess(
             base_img = np.array(PIL.Image.open(base_img))
         elif isinstance(base_img, np.ndarray):
             base_file = processing_utils.save_img_array_to_cache(
-                base_img, cache_dir=self.GRADIO_CACHE
+                base_img, cache_dir=self.GRADIO_CACHE, format=self.format
             )
             base_img_path = str(utils.abspath(base_file))
         elif isinstance(base_img, PIL.Image.Image):
             base_file = processing_utils.save_pil_to_cache(
-                base_img, cache_dir=self.GRADIO_CACHE
+                base_img, cache_dir=self.GRADIO_CACHE, format=self.format
```

Comment:
We need to update the type hint of `format` in `save_pil_to_cache` to allow for an arbitrary string